packages/shared/src/components/cards/ActionButtons.tsx

```diff
@@ -157,19 +158,15 @@ export default function ActionButtons({
         </SimpleTooltip>
         {insaneMode && lastActions}
       </ConditionalWrapper>
-      {insaneMode ? (
+      {insaneMode && !isInternalReadType(post) ? (
         <div
           className={classNames('flex justify-between', visibleOnGroupHover)}
         >
           <ReadArticleButton
             className="mr-2 btn-primary"
-            href={
-              post.type === PostType.Share
-                ? post.sharedPost.permalink
-                : post.permalink
-            }
+            href={getReadArticleLink(post)}
             onClick={onReadArticleClick}
-            openNewTab={openNewTab}
+            openNewTab={isSharedPostSquadPost(post) ? false : openNewTab}
```

Comment:
I think this can be simplified to
```suggestion
            openNewTab={!isSharedPostSquadPost(post) && openNewTab}
```